contracts/Registry/TokenCuratedRegistry.sol

```diff
@@ -1,59 +1,107 @@
 pragma solidity ^0.4.24;
 
+import './LockableItemRegistry.sol';
 import './StakedRegistry.sol';
-import '../Challenge/ChallengeFactory.sol';
-import 'openzeppelin-solidity/contracts/token/ERC20/ERC20.sol';
+import '../Challenge/IChallengeFactory.sol';
+import '../Challenge/IChallenge.sol';
 
-contract TokenCuratedRegistry is StakedRegistry {
-  uint public applicationPeriod;
-  ChallengeFactory public challengeFactory;
+contract TokenCuratedRegistry is StakedRegistry, LockableItemRegistry {
+  uint applicationPeriod;
+  IChallengeFactory public challengeFactory;
+  mapping(bytes32 => IChallenge) public challenges;
+  
+  constructor(ERC20 _token, uint _minStake, uint _applicationPeriod, IChallengeFactory _challengeFactory)
+  StakedRegistry(_token, _minStake)
+  public {
+    applicationPeriod = _applicationPeriod;
+    challengeFactory = _challengeFactory;
+  }
 
-  modifier itemNotLockedInChallenge(bytes32 id) {
-    require(!inChallengePhase(id));
-    _;
+  // Adds an item to the `items` mapping, transfers token stake from msg.sender, and locks
+  // the item from removal until now + applicationPeriod.
+  function add(bytes32 data) public returns (bytes32 id) {
+    id = super.add(data);
+    unlockTimes[id] = now.add(applicationPeriod);
   }
 
-  mapping(bytes32 => ItemCurationData) itemsCurationData;
+  // Removes an item from the `items` mapping, and deletes challenge state. Requires that
+  // there is not an active or passed challenge for this item. OwnedItemRegistry.remove
+  // requires that this is called by the item owner. LockableItemRegistry.remove requires
+  // that the item is not locked.
+  function remove(bytes32 id) public {
+    if (challengeExists(id)) {
+      require(challengeEnded(id) && !challengePassed(id));
+    }
+    delete challenges[id];
+    super.remove(id);
+  }
 
-  struct ItemCurationData {
-    uint applicationExpiry;
-    bool whitelisted;
-    address challengeAddress;
-    bool challengeResolved;
+  // Creates a new challenge for an item.
+  // Requires that the item exists, and that there is no existing challenge for the item.
+  // Requires msg.sender (the challenger) to match the owner's stake by transferring to
+  // this contract. The challenger's and owner's stake is transferred to the newly created
+  // challenge contract.
+  function challenge(bytes32 id) public {
```

Comment:
in the vanilla version, challenge should auto-pass if the `ownerStake` is less than `minStake` (which may have changed since item was added)